File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        inp = "([{"
        oup = ")]}"
        for i in s:
            if len(stack)>0 and (i in oup and stack[-1]==inp[oup.index(i)]):
                stack.pop()
            else:
                stack.append(i)
        return True if len(stack)==0 else False

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<s.length();i++){
            Character tmp = s.charAt(i);
            if(!stack.isEmpty() && ((tmp.equals('}') && stack.peek().equals('{')) || (tmp.equals(')') && stack.peek().equals('(')) || (tmp.equals(']') && stack.peek().equals('[')))) stack.pop();
            else{
                stack.push(tmp);
            }
        }
        return stack.isEmpty();
    }
}